packages/trace-viewer/src/sw/snapshotRenderer.ts

```diff
@@ -249,6 +272,10 @@ function snapshotScript(...targetIds: (string | undefined)[]) {
     const targetElements: Element[] = [];
     const canvasElements: HTMLCanvasElement[] = [];
 
+    let topFrameWindow: Window = window;
+    while (topFrameWindow !== topFrameWindow.parent && !topFrameWindow.location.pathname.match(/\/page@[a-z0-9]+$/))
```

Comment:
It may not be the actual top window. In fact, it's only the top window when we open the snapshot in a new tab. In trace viewer / UI mode, it must be the snapshot root window, which is an iframe:

![image](https://github.com/user-attachments/assets/864dc96f-06fa-4a71-b51a-3e692b1f9665)

Maybe `topFrameWindow` should be called `topSnapshotWindow` instead.